flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializerSnapshot.java

```diff
@@ -56,6 +55,9 @@
     /** Contains the actual content for the serializer snapshot. */
     private PojoSerializerSnapshotData<T> snapshotData;
 
+    /** Configuration of the current execution. */
+    private ExecutionConfig executionConfig;
```

Comment:
This is a serious issue :(

We should not do that ever! This gives an impression the `executionConfig` is part of the serialized snapshot which it **must** not!

If I understand correctly this is necessary, because the method `resolveSchemaCompatibility(TypeSerializerSnapshot<T> oldSerializerSnapshot` is called at a point where the originating `newSerializer` has not been used yet and thus has not registeredSubclassSerializers.

We need to fix it in a different way. The current approach makes it even messier than it already is and we can not go with that approach.

I'll give it a thought.

Reply:
After thinking over it for a while I somewhat softened my opinion here.

Yes, it is messy, but I think we're forced to do it that way :( Do you mind adding a big, strong warning next to the `executionConfig` that it should never, under any circumstances end up serialized?

Reply:
Thanks for the reminder.
I have added a comment above the field.
Please help me check again whether it fits your idea, Thanks.